src/Header/HttpListener.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef GROUP4_HEADER_HTTPLISTENER_H_
+#define GROUP4_HEADER_HTTPLISTENER_H_
+
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <netinet/in.h>
+#include <memory.h>
+
+#include <thread>
+
+#include "MainManager.h"
+
+class MainManager
+{
+};
+
+//Listen specific port and will callback MainManager function when receive request
+class HttpListener
+{
+
+public:
+    //Singleton code convention, must bind callback for listening port and require listening port
+    static std::unique_ptr<HttpListener> GetInstance(void (*callbackFunction)(std::string), const int &portNum);
```

Comment:
Please create a separate InitInstance() for the singleton that requires initialization.
Use assert() if GetInstance() is called before InitInstance.

Reply:
Also, unique_ptr doesn't work with singleton.

Reply:
Done